numpy/lib/function_base.py

```diff
@@ -1622,21 +1622,29 @@ def trim_zeros(filt, trim='fb'):
     [1, 2]
 
     """
-    first = 0
-    trim = trim.upper()
-    if 'F' in trim:
-        for i in filt:
-            if i != 0.:
-                break
-            else:
-                first = first + 1
-    last = len(filt)
-    if 'B' in trim:
-        for i in filt[::-1]:
-            if i != 0.:
-                break
-            else:
-                last = last - 1
+    arr = np.asanyarray(filt).astype(bool, copy=False)
+
+    if arr.ndim != 1:
+        raise ValueError('trim_zeros requires an array of exactly one dimension')
+    elif not len(arr):
+        return filt
+
+    trim_upper = trim.upper()
+    len_arr = len(arr)
+    first = last = None
+
+    if 'F' in trim_upper:
+        first = arr.argmax()
+        # If `arr[first] is False` then so are all other elements
+        if not arr[first]:
+            return filt[len_arr:]
+
+    if 'B' in trim_upper:
+        last = len_arr - arr[::-1].argmax()
+        # If `arr[last - 1] is False` then so are all other elements
+        if not arr[last - 1]:
+            return filt[len_arr:]
```

Comment:
```suggestion
            return filt[:0]
```
would be slightly truer to intent here, "trim the back of the array"

Reply:
I agree; implemented in https://github.com/numpy/numpy/pull/16911/commits/d75061d6cbf32575a30dfe3628d38e073a9220d4.